components/Weather.js

```diff
@@ -33,72 +54,75 @@ export default class Weather extends React.Component {
           // See error code charts below.
           console.log(error.code, error.message);
         },
-        {enableHighAccuracy: true, timeout: 15000},
+        { enableHighAccuracy: true, timeout: 15000 },
       );
     }
   };
-  _parseWeatherData = (data) => {
+
+  parseWeatherData = (data) => {
     const weather = {};
-    weather.city = datas.city.name;
+    weather.city = data.city.name;
     weather.dataPerHour = [];
-    for (const data of datas.list) {
+    data.forEach((weatherLog) => {
       const perHour = {
-        temp: parseInt(data.main.temp),
-        humidity: data.main.humidity,
-        description: data.weather[0].description,
-        wind: data.wind.speed,
-        hour: data.dt_txt.substr(-8, 5),
-        date: data.dt,
-        icon: data.weather[0].icon,
+        temp: parseInt(weatherLog.main.temp, 10),
+        humidity: weatherLog.main.humidity,
+        description: weatherLog.weather[0].description,
+        wind: weatherLog.wind.speed,
+        hour: weatherLog.dttxt.substr(-8, 5),
+        date: weatherLog.dt,
+        icon: weatherLog.weather[0].icon,
       };
       weather.dataPerHour.push(perHour);
```

Comment:
You should use map instead of for each here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map